neuropoly/spinalcordtoolbox
# sct_deepseg_sc, sct_deepseg_lesion: Allow to input image with one axial slice

**Goal:** Allow to input image with only one axial slice.

**Issue:** `isct_spine_detect` requires ```nz>1```.

**Solution:** If ```nz==1```, then ask the user to use `-centerline cnn` instead of `-centerline svm` (default).

To test:
```
cd sct_example_data/t2s
sct_crop_image -i t2s.nii.gz -dim 2 -start 2 -end 2 -o t2s_Z010.nii.gz
sct_deepseg_sc -i t2s_Z010.nii.gz -c t2s
```
Raises an error message:
```
Your image contains only one axial slice, please re-run the function using -centerline cnn.
```
Then,
```
sct_deepseg_sc -i t2s_Z010.nii.gz -c t2s -centerline cnn
```

Same for `sct_deepseg_lesion`.

Note that I have remove some duplicated code: The centerline detection is now done by a function in `sct_deepseg_sc` named `find_centerline`, which is called by `sct_deepseg_lesion`.
